### PR TITLE
Create prereqs.md

### PR DESCRIPTION
**Pull request by blackdwarf**
Creating an overall .NET Core pre-req native deps file to be able to point people to.

/cc @richlander @gkhanna79 @ellismg

### CODE REVIEW

File: Documentation/prereqs.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+On Windows, the only dependency is the VC++ Redistributable. Depending on the version of Windows you are running on, the versions are changing.
+
+> **Note:** these dependencies are chained in the installer, you only need to install them manually if you are using
+> the installer script (`dotnet-install.sh` or `dotnet-install.ps1`). 
+
+* Windows 10
+    * [Visual C++ Redistributable for Visual Studio 2015](https://www.microsoft.com/en-us/download/details.aspx?id=48145)
+* Pre-Windows 10
+    * Please make sure that your Windows installation is up-to-date and includes hotfix [KB2533623] (https://support.microsoft.com/en-us/kb/2533623) installed through Windows Update.
+    * [Visual C++ Redistributable for Visual Studio 2012 Update 4](https://www.microsoft.com/en-us/download/confirmation.aspx?id=30679)
```
Comment (gkhanna79) [inline Documentation/prereqs.md]: Why is this required?

CC @schellap

**Comment by gkhanna79** [comment]
@weshaggard PTAL as well.

File: Documentation/prereqs.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
+.NET Core native prerequisites
+==============================
+
+This document outlines the dependencies needed to run .NET Core tools and applications. 
+
+## Windows dependencies
+On Windows, the only dependency is the VC++ Redistributable. Depending on the version of Windows you are running on, the versions are changing.
```
Comment (gkhanna79) [inline Documentation/prereqs.md]: CRT Dependency: 

On Pre-Win10 OS, the only dependency is uCRT Windows update that can be installed from https://www.microsoft.com/en-us/download/details.aspx?id=48234. You can point folks to https://blogs.msdn.microsoft.com/vcblog/2015/03/03/introducing-the-universal-crt/ for details. 

On Win10+, this is available out of the box.

OS Dependency:

You should call out that for curated OS SKUs (e.g. the ones that are not full server or client - like Nano), they need to have Reverse Forwarders in their OS image.

File: Documentation/prereqs.md (hunk 1 above)
Comment (blackdwarf) [inline Documentation/prereqs.md]: I don't know if it is. Could be a remnant of something else. Though I'm pretty sure @leecow ran into something similar recently on Win7 vanilla machine.

File: Documentation/prereqs.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
+.NET Core native prerequisites
+==============================
+
+This document outlines the dependencies needed to run .NET Core tools and applications. 
+
+## Windows dependencies
```
Comment (gkhanna79) [inline Documentation/prereqs.md]: Where are we calling out the list of supported OS platforms and architectures since that is part of the pre-req as well?

Comment (blackdwarf) [inline Documentation/prereqs.md]: https://github.com/dotnet/core/blob/master/roadmap.md here, I will link, that's a good point.

File: Documentation/prereqs.md (hunk 2 above)
Comment (blackdwarf) [inline Documentation/prereqs.md]: WTF are "reverse forwarders"??

File: Documentation/prereqs.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+* gettext 
+* libcurl-devel 
+* openssl-devel 
+* zlib 
+* libicu-devel
+
+## OS X 
+OS X requires the following libraries and versions installed:
+
+* libssl >= 1.1
```
Comment (bartonjs) [inline Documentation/prereqs.md]: >= 1.0.1.

The current version from homebrew is 1.0.2, and 1.1 isn't quite released yet, and may not even be compatible.

File: Documentation/prereqs.md (hunk 1 above)
Comment (leecow) [inline Documentation/prereqs.md]: Without the 2012 crt, dotnet build will toss an error that it cannot find Microsoft.DiaSymReader.Native.amd64.dll.

File: Documentation/prereqs.md (hunk 2 above)
Comment (gkhanna79) [inline Documentation/prereqs.md]: Also, see:

https://social.technet.microsoft.com/Forums/en-US/e08daebc-c298-4932-a9d7-ff08fabf9536/nano-server-packages?forum=NanoServer

Some formal definition at https://blogs.technet.microsoft.com/windowsserver/2015/11/16/moving-to-nano-server-the-new-deployment-option-in-windows-server-2016/
